azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/NetworkSecurityGroupApiLiveTest.java

```diff
@@ -0,0 +1,135 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import com.google.common.base.Predicate;
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityGroup;
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityGroupProperties;
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityRule;
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityRuleProperties;
+
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityRuleProperties.Access;
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityRuleProperties.Direction;
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityRuleProperties.Protocol;
+import org.jclouds.azurecompute.arm.functions.ParseJobStatus;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+
+import org.jclouds.util.Predicates2;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.net.URI;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.AssertJUnit.assertNull;
+
+@Test(groups = "live", singleThreaded = true)
+public class NetworkSecurityGroupApiLiveTest extends BaseAzureComputeApiLiveTest {
+
+   private String resourcegroup;
+   private static String DEFAULT_NSG_NAME = "testNetworkSecurityGroup";
+
+   private NetworkSecurityGroup createGroup() {
+      NetworkSecurityRule rule = NetworkSecurityRule.create("denyallout", null, null,
+                     NetworkSecurityRuleProperties.create("deny all out",
+                                                 Protocol.Tcp,
+                                                 "*",
+                                                 "*",
+                                                 "*",
+                                                 "*",
+                                                 Access.Deny,
+                                                 4095,
+                                                 Direction.Outbound));
+      ArrayList<NetworkSecurityRule> ruleList = new ArrayList<NetworkSecurityRule>();
+      ruleList.add(rule);
+      NetworkSecurityGroup nsg = NetworkSecurityGroup.create("samplensg", "westus", null,
+                                                             NetworkSecurityGroupProperties.create(ruleList, null,
+                                                                                                   null, null, null,
+                                                                                                   null),
+                                                             null);
```

Comment:
Same here, providing a builder would help code be more readable and get rid of all those null values.